src/workspace/resolver.rs

```diff
@@ -689,6 +792,67 @@ impl WorkspaceResolver {
   }
 }
 
+#[derive(Deserialize, Serialize)]
+pub struct SerializedWorkspaceResolverImportMap<'a> {
+  #[serde(borrow)]
+  pub specifier: Cow<'a, str>,
+  #[serde(borrow)]
+  pub json: Cow<'a, str>,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct SerializedResolverWorkspaceJsrPackage<'a> {
+  #[serde(borrow)]
+  pub relative_base: Cow<'a, str>,
+  #[serde(borrow)]
+  pub name: Cow<'a, str>,
+  pub version: Cow<'a, Option<Version>>,
+  pub exports: Cow<'a, IndexMap<String, String>>,
+}
+
+#[derive(Deserialize, Serialize)]
+pub struct SerializableWorkspaceResolver<'a> {
+  #[serde(borrow)]
+  pub import_map: Option<SerializedWorkspaceResolverImportMap<'a>>,
+  #[serde(borrow)]
+  pub jsr_pkgs: Vec<SerializedResolverWorkspaceJsrPackage<'a>>,
+  pub package_jsons: BTreeMap<Cow<'a, str>, serde_json::Value>,
+  pub pkg_json_resolution: PackageJsonDepResolution,
+}
+
+#[derive(Debug, Clone)]
+pub struct BaseUrl<'a>(Cow<'a, Url>);
```

Comment:
Maybe we can make this private and only have `Url` accepted in the methods above?

Reply:
Yep, much better.